internal/encoding/asn1/asn1.go

```diff
@@ -0,0 +1,219 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package asn1 decodes BER-encoded ASN.1 data structures and encodes in DER.
+// Note: DER is a subset of BER.
+// Reference: http://luca.ntop.org/Teaching/Appunti/asn1.html
+package asn1
+
+import (
+	"bytes"
+	"encoding/asn1"
+)
+
+// Common errors
+var (
+	ErrBytesAtTheEnd           = asn1.StructuralError{Msg: "invalid bytes at the end of the BER data"}
+	ErrEarlyEOF                = asn1.SyntaxError{Msg: "early EOF"}
+	ErrInvalidBerData          = asn1.StructuralError{Msg: "invalid BER data"}
+	ErrInvalidOffset           = asn1.StructuralError{Msg: "invalid offset"}
+	ErrInvalidSlice            = asn1.StructuralError{Msg: "invalid slice"}
+	ErrUnsupportedLen          = asn1.StructuralError{Msg: "length method not supported"}
```

Comment:
`Length` should be named `Length` instead of `Len` unless it is for the same functionality of `len()`
```suggestion
	ErrUnsupportedLength       = asn1.StructuralError{Msg: "length method not supported"}
```

Reply:
Updated.